pkg/config/validation/validation.go

```diff
@@ -2456,23 +2458,38 @@ var ValidateServiceEntry = registerValidateFunc("ValidateServiceEntry",
 				ipAddr := net.ParseIP(endpoint.Address) // Typically it is an IP address
 				if ipAddr == nil {
 					if err := ValidateFQDN(endpoint.Address); err != nil { // Otherwise could be an FQDN
-						errs = appendErrors(errs,
+						errs = appendValidation(errs,
 							fmt.Errorf("endpoint address %q is not a valid FQDN or an IP address", endpoint.Address))
 					}
 				}
-				errs = appendErrors(errs,
+				errs = appendValidation(errs,
 					labels.Instance(endpoint.Labels).Validate())
 				for name, port := range endpoint.Ports {
 					if !servicePorts[name] {
-						errs = appendErrors(errs, fmt.Errorf("endpoint port %v is not defined by the service entry", port))
+						errs = appendValidation(errs, fmt.Errorf("endpoint port %v is not defined by the service entry", port))
 					}
-					errs = appendErrors(errs,
+					errs = appendValidation(errs,
 						ValidatePortName(name),
 						ValidatePort(int(port)))
 				}
 			}
+			if len(serviceEntry.Addresses) > 0 {
+				var hasTCPPort bool
+				for _, port := range serviceEntry.Ports {
+					p := protocol.Parse(port.Protocol)
+					if p.IsTCP() {
+						hasTCPPort = true
+						break
+					}
+				}
+				if hasTCPPort && len(serviceEntry.Hosts) > 1 {
+					// TODO: prevent this invalid setting, maybe in 1.11+
+					errs = appendValidation(errs, WrapWarning(fmt.Errorf("service entry can not have more than one host specified "+
```

Comment:
Do we want to fail now with a feature flag default to true (fail) especially if our longer term goal is to fail. We can highlight this in release notes and people can disable this if they have service entries that fail this validation.

WDYT?

Reply:
With this warning thrown, user should change accordingly. This behaves like k8s API deprecation note. I am concerned about the abuse of feature flag to control such little functionality.

Reply:
Sure. Should we release note this so that people are aware of this?

Reply:
> Do we want to fail now with a feature flag default to true

I do not think so, we should never have a backwards incompatible change like this. It will break users on upgrade.

If we do want to do this, please bring it up in TOC (I can proxy for you if its too late timezone wise, although I don't personally think we should do it)

Reply:
Basically, we should never ever reject a config that was previously valid.

Reply:
I agree we should not have backward incompatible change but given that it is invalid - Do we support it forever? Or do we advertise it via release notes that we are giving a warning now and this may become invalid in future like how Envoy is dealing with its deprecations?

Reply:
Its better to warn and do nothing than to reject it, if we have to. Rejecting means users are completely blocked from upgrade.

Note that the config is not invalid - it does *something*. It just is weird

Reply:
Then what should we do now? Is this ok

Reply:
I am fine with just warning. Let us land this.